### PR TITLE
fix(formatter): Enable calling parseTestCaseAttempt on test cases that haven't completed

### PR DESCRIPTION
**Pull request by jan-molak**
In [Serenity/JS Cucumber formatter](https://serenity-js.org/modules/cucumber) I need to be able to invoke `parseTestCaseAttempt` when Cucumber emits the `testStepStarted` message.

However, the `parseStep` function used internally by `parseTestCaseAttempt` assumes that it's invoked on a step that has already finished running and therefore has a `result`, and that this `result` has a `status`.

 Invoking `parseTestCaseAttempt` on a test case that has unfinished steps causes an "NPE" thrown [here](https://github.com/cucumber/cucumber-js/blob/4ec4a98413488d6f516b735277a001b783806e5b/src/formatter/helpers/test_case_attempt_parser.ts#L116).

This PR addresses this issue by defaulting any `undefined` step results to an `UNKNOWN` status.
Since all the properties of `ITestStepResult` are optional `{ status: Status.UNKNOWN }` seems like a sufficient dummy.

I haven't found any unit tests in this area and all the high-level tests are still passing.

@charlierudolph @davidjgoss - please let me know if you're happy with this PR or if there are any changes you'd like to see before it could be merged and released.

Thanks!
Jan

### CODE REVIEW

**Comment by charlierudolph** [comment]
Code changes makes sense to me. Can you please add a test for this? Fine with unit or feature. Depends on how you ran into this / would like to test it.

**Comment by aurelien-reeves** [comment]
The code makes sense to me too.
@jan-molak would you have the possibility to add a test for this?

**Comment by aurelien-reeves** [comment]
@jan-molak as we did not had news since several month, I took the liberty to complete your PR. Still, you will remain the author for this!

Does it look as you expected?

**Comment by jan-molak** [comment]
Hey @aurelien-reeves, your changes look great, thanks for finishing it all up. Much appreciated!

File: src/formatter/helpers/test_case_attempt_parser_spec.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        const testCaseAttempt = {
+          gherkinDocument: parsed.gherkinDocument,
+          pickle: parsed.pickles[0],
+          testCase,
+          attempt: 0,
+          willBeRetried: false,
+          stepAttachments: {},
+          stepResults: {},
+          worstTestStepResult: new TestStepResult(),
+        }
```
Comment (charlierudolph) [inline src/formatter/helpers/test_case_attempt_parser_spec.ts:61]: This is a lot of mocking out. Thoughts on creating a testCaseAttempt in the same way as the other and then setting stepResults to an empty object?

**Review by charlierudolph — APPROVED** [review]

**Pull request merged**